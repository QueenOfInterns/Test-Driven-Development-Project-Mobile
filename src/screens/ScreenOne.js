/* eslint-disable react-native/no-inline-styles */
import React, {Component} from 'react';
import {View, Text} from 'react-native';
import {connect} from 'react-redux';
import {addOne} from '../state/actions/addOne';

class ScreenOne extends Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    console.log('count', this.props.counter);
  }

  render() {
    return (
      <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
        <Text>Hello World!</Text>
      </View>
    );
  }
}

const mapStateToProps = state => {
  return {
    counter: state.testReducer.counter,
  };
};

const mapDispatchToProps = {
  Increment: addOne,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ScreenOne);
