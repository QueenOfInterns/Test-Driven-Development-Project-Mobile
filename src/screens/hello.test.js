/* eslint-disable no-undef */
/* eslint-disable no-unused-vars */
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ScreenOne from './ScreenOne.js';

describe('Hello World Screen', () => {
  describe('The code for the test', () => {
    it('renders correctly', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<ScreenOne />);
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });
});
