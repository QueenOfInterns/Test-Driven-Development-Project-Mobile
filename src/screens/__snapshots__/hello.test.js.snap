// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello World Screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Hello World!
  </Text>
</View>
`;
